R/t-distribution

```diff
@@ -0,0 +1,270 @@
+  theme = bslib::bs_theme(version = 4),
+  uiOutput("css"),
+  withMathJax(),
+  miniContentPanel(
+    conditionalPanel(
+      "output.help",
+      div(
+        class = "alert alert-info p-1 rounded d-flex align-items-center",
+        icon("warning", class = "h3 mx-3", verify_fa = FALSE),
+        
+        p(
+          class = "mb-0",
+          "This app has three modes: critical values, p-values, and combined. You can select the mode by adding",
+          code("?mode=critical_values"),
+          ",",
+          code("?mode=p_values"),
+          ", or",
+          code("?mode=combined"),
+          "to the url. You are currently viewing the combined mode. Choosing a mode will suppress this message."
+        )
+      )
+    ),
+    plotOutput("distribution", height = "100%")
+  ),
+  miniButtonBlock(
+    style = "justify-content: space-around; gap: 1em; padding: .5em;",
+    # critical values
+    conditionalPanel( # set significance level
+      "output.mode == 'combined' || output.mode == 'critical_values'",
+       div(
+         id = "critical-value-panel",
+         class = "d-flex flex-column",
+         
+         div(# set alpha
+          class = "d-flex",
+          sliderInput("alpha", "Type I error, \\(\\alpha\\)", 0.01, 0.2, 0.05, 0.01)
+          ),
+         div(# degree of freedom
+          class = "align-items-center",
+          sliderInput(inputId = "df",
+                      label = "Degrees of Freedom", min = 1, max = 200, value = 1, step = 1)
+                     
+          # numericInput(inputId = "df", label = "Degrees of Freedom", STARTING_DF, 
+          #              min = 1, max = 200, value = 1, step = 1
+          #              )
+          ),
+      ),
+    ),
+    
+    conditionalPanel( 
+      "output.mode == 'combined' || output.mode == 'p_values'",
+      div( # set t value
+        id = "p-value-panel",
+        class = "d-flex",
+        div(
+          class = "d-flex align-items-center",
+          numericInput("t_value", "\\(t\\)-value", STARTING_T_VALUE, -4, 4, 0.01), 
+          actionButton(
+            "t_to_p",
+            "Calculate \\(p\\)-value",
+            class = "btn-primary",
+            style = "position: relative; top: 8px;"
+          )
+        ),
+        div( # calculate p_value
+          class = "d-flex align-items-center",
+          numericInput("p_value", "\\(p\\)-value", STARTING_P_VALUE, 0, 1, 0.01), 
+          actionButton(
+            "p_to_t",
+            "Calculate \\(t\\)-value",
+            class = "btn-primary",
+            style = "position: relative; top: 8px;"
+          )
+        )
+      )
+    )
+  )
+)
+
+# SERVER ---------------------------------------------------------
+server <- function(input, output, session) {
+  crit_lower <- reactive(qt(input$alpha / 2, input$df)) # find lower critical value
+  crit_upper <- reactive(qt(1 - input$alpha / 2, input$df)) # find upper critical value
+  data <- reactive(tibble(
+    t = c(
```

Comment:
re; adaptive range, check if critical t falues fall outside of range, and if necessary, extend the range

Reply:
we'll need to see if that has implications for the precision, as for df=1 the range is ~-25,25 or so